packages/agent-toolkit/src/core/tools/platform-api-tools/get-board-activity/get-board-activity-tool.ts

```diff
@@ -0,0 +1,81 @@
+import { z } from 'zod';
+import {
+  GetBoardAllActivityQuery,
+  GetBoardAllActivityQueryVariables,
+} from '../../../../monday-graphql/generated/graphql';
+import { getBoardAllActivity } from './graphql';
+import { ToolInputType, ToolOutputType, ToolType } from '../../../tool';
+import { BaseMondayApiTool, createMondayApiAnnotations } from './../base-monday-api-tool';
+import { TIME_IN_MILLISECONDS } from '../../../../utils';
+
+export const getBoardActivityToolSchema = {
+  boardId: z.number().describe('The id of the board to get activity for'),
+  fromDate: z
+    .string()
+    .optional()
+    .describe('Start date for activity range (ISO8601DateTime format). Defaults to 30 days ago'),
+  toDate: z.string().optional().describe('End date for activity range (ISO8601DateTime format). Defaults to now'),
+};
+
+export class GetBoardActivityTool extends BaseMondayApiTool<typeof getBoardActivityToolSchema | undefined> {
+  name = 'get_board_activity';
+  type = ToolType.READ;
+  annotations = createMondayApiAnnotations({
+    title: 'Get Board Activity',
+    readOnlyHint: true,
+    destructiveHint: false,
+    idempotentHint: true,
+  });
+
+  private defaultLimit = 1000;
+
+  getDescription(): string {
+    return 'Get board activity logs for a specified time range (defaults to last 30 days)';
```

Comment:
@RomKadria I kept this comment with the 30 days as it might give better indication to agents

Reply:
The only indication they will have is about the 30 days, the agent shouldn't care he sees this tool to know the data he gets. for the how long -> this is params discovery